### ISSUE TITLE
📎 Handle `PromiseLike` in `noFloatingPromises`

### ISSUE THREAD

**Issue by arendjr** [biomejs/biome#6604]
### Description

`noFloatingPromises` should detect `PromiseLike` as well:

```ts
async function promiseLike(): PromiseLike<number> {
	return new Promise.resolve(1);
}

promiseLike();
```

We may want to implement #5977 first, since we may get a `PromiseLike` definition from that.
